mLRS/Common/time_stats.h

```diff
@@ -0,0 +1,83 @@
+//*******************************************************
+// Copyright (c) MLRS project
+// GPL3
+// https://www.gnu.org/licenses/gpl-3.0.de.html
+// OlliW @ www.olliw.eu
+//*******************************************************
+// time_stats
+//*******************************************************
+#ifndef TIME_STATS_H
+#define TIME_STATS_H
+#pragma once
+
+// Time Statistics for up to PM_NUM_BLOCK pairs of measurement points
+#ifdef USE_DEBUG
+#define TS_NUM_BLOCK 10
+
+struct TS_store {
+    uint32_t start;
+    uint32_t last_report;
+    uint32_t total;
+    uint32_t min;
+    uint32_t max;
+    uint32_t count;
+    uint32_t overflow;
+    uint16_t last_cnt;
+};
+static struct TS_store TS_data[TS_NUM_BLOCK];
+
+inline uint32_t TS_MICROS32(int block) {
+
+    // Separate overflow and last_cnt for each block allows either ISR/normal use, but not both
+    uint16_t cnt = MICROS_TIMx->CNT;
```

Comment:
I wanted to avoid the extra function call overhead.  But you have reminded me I meant to make this work for ESP as well.  Can we mark micros16() as inline in both esp-timer.h and timer.h?

Reply:
with inline or not the compiler will do anyway what it wants. I'm going more an more away from it.

Reply:
I guess if we want to force the compiler to inline, we can also add __attribute__((always_inline)).  Is there any reason not to do this for micros16() in esp-timer.h and timer.h?

Reply:
my point rather was I don't want to force to inline. I tended to rather go away form that since the actual effect is usually like zero.